### DISCUSSION TITLE
Documentation for new endpoint changes

### DISCUSSION THREAD

**Discussion by clara-j** [teslamate-org/teslamate#3959] [General]
I see the PR for custom endpoints has been merged, and it seems like the new variables were added to the advanced section for environment_variables.  But I am not seeing how to use those variables for Teslemetry (which I have signed up for) or MyTeslaMate.

I can help with writing the documentation but not sure where to start to pull the details needed.  If anyone can point me at any PR/Issue comments may have some of the details that would be helpful.

I was thinking a new page under guides (https://docs.teslamate.org/docs/guides/) to document this with the follow.
* Summary of why this is needed (changes to current API)
* Impact/limitations of new endpoints.  I'm not clear on this yet myself but seems like there is some
* Guide for both services to explain what values need to be populated and where to get them

**Comment by JakobLichterfeld** [comment]
@jlestel if you want to shoot some breadcrumbs @clara-j want to improve docu on endpoint env variables

**Reply by jlestel** [reply]
@clara-j  I initiated the PR #3978 to add this guide.

You can see the result here: https://deploy-preview-3978--teslamate.netlify.app/docs/guides/api

Let me know if you can try it with Teslemetry, I don't have a paid account.

And all comments are welcome.

**Reply by clara-j** [reply]
This is great, I read through and seems exactly like I was looking for.  I am not able to test tonight but will try to do it tomorrow when I get home.

One question, I see mentioned but not explicitly outlined is what is the implications of disabling streaming with Teslemetry.  I have been reading old posts around disabling streaming api with the legacy API and seems like ihas gaps when the car is going to sleep, and could impact sleeping (https://github.com/teslamate-org/teslamate/discussions/1288#discussioncomment-491506).  Does this same thing happen with Teslemetry?  Or is there anything else the not having streaming enabled cause?

**Reply by clara-j** [reply]
I followed the instructions to add the items to the docker file and did a restart and still seeing the details for my car on the TeslaMate UI so as far as I can see all seems to be working.  And was very easy to follow.

I would just want to know the impact of disabling the streaming.

**Reply by jlestel** [reply]
> all seems to be working. And was very easy to follow.

wow on the first try :D

> I would just want to know the impact of disabling the streaming.

I don't know the full implications of operating without streaming.

What I can tell you is that as the number of calls to the Fleet API is limited daily, you should see 429 errors appear fairly quickly. (owner API users also saw these 429 issues last week [#3957](https://github.com/teslamate-org/teslamate/issues/3957))

I made a PR about this to limit the number of calls #3940. So we have to wait for this PR to limit 429 and be more "stable". 

Unfortunately, it remains quite limited: streaming (with Telemetry) is the only way to have fresh data with new Tesla APIs. Ideally, we could even retrieve more fields with Telemetry (up to 20) so that this data is always updated.
